error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -1081,7 +1081,8 @@ static final class MonoSingleOptional<T> {
     Mono<Optional<T>> before(Mono<T> mono) {
       return Refaster.anyOf(
           mono.flux().collect(toOptional()),
-          mono.map(Optional::of).defaultIfEmpty(Optional.empty()),
+          mono.map(Optional::of),
+          mono.singleOptional().defaultIfEmpty(Optional.empty()),
```

Comment:
`Mono#singleOptional` will default to `Optional.empty()` anyways, so keeping that in mind we can break this rule into two:
1. Re-write `mono.map(Optional::of)`
2. Omit the unnecessary `.defaultIfEmpty(Optional.empty()` when possible.

The existing case would still be re-written by applying to (1) and then (2).

Reply:
Looks like we can even generalize this to any `defaultIfEmpty` and `switchIfEmpty` argument :)

Reply:
:bulb: Since the `Mono` will never be empty. Nice, I didn't even consider this.

Reply:
This is why we have code review :smile: `:team-work:`